addon/services/hifi.js

```diff
@@ -188,17 +188,20 @@ export default Service.extend(Ember.Evented, DebugLogging, {
     });
 
 
-    promise.then(({sound}) => sound.set('metadata', options.metadata));
+    promise.then(({sound}) => sound.set('metadata', (options.metadata || {})));
     promise.then(({sound}) => this.get('soundCache').cache(sound));
 
-
     // On audio-played this pauses all the other sounds. One at a time!
     promise.then(({sound}) => this.get('oneAtATime').register(sound));
 
     promise.then(({sound}) => sound.on('audio-played', () => {
       let previousSound = this.get('currentSound');
       let currentSound  = sound;
 
+      if (previousSound && get(previousSound, 'isPlaying')) {
+        this.trigger('current-sound-interrupted', previousSound);
```

Comment:
Oh, true. I didn't test for that. I guess we could wrap it in an Ember.run.once? There's no exclusion currently but it needs to be in both places. Nice catch.